ui/src/app/machines/views/Machines.js

```diff
@@ -1,72 +1,139 @@
+          title={
+            <>
+              <Row>
+                <Col size="10">
+                  <ul className="p-inline-list">
+                    <li className="p-inline-list__item p-heading--four">
+                      Machines
+                    </li>
+                    {machinesLoaded ? (
+                      <li
+                        className="p-inline-list__item u-text--light"
+                        data-test="machine-count"
+                      >
+                        {`${machines.length} ${pluralize(
+                          "machine",
+                          machines.length
+                        )} available`}
+                      </li>
+                    ) : (
+                      <Loader
+                        className="u-no-padding u-no-margin"
+                        inline
+                        text="Loading..."
+                      />
+                    )}
+                  </ul>
+                </Col>
+                <Col size="2" className="u-align--right">
+                  <Route exact path="/machines">
+                    <ContextualMenu
+                      data-test="add-hardware-dropdown"
+                      hasToggleIcon
+                      links={getAddHardwareLinks(navigationOptions)}
+                      position="right"
+                      toggleAppearance="neutral"
+                      toggleLabel="Add hardware"
+                    />
+                  </Route>
+                </Col>
+              </Row>
+              <Row>
+                <Col size="12">
+                  <hr className="u-no-margin--bottom" />
+                  <Tabs
+                    data-test="machine-list-tabs"
+                    links={[
+                      {
+                        active: location.pathname === "/machines",
+                        label: `${
+                          machinesLoaded ? `${machines.length} ` : ""
+                        }${pluralize("Machine", machines.length)}`,
+                        path: "/machines"
+                      },
+                      {
+                        active: location.pathname === "/pools",
+                        label: `${
+                          resourcePoolsLoaded ? `${resourcePools.length} ` : ""
+                        }${pluralize("Resource pool", resourcePools.length)}`,
+                        path: "/pools"
+                      }
+                    ]}
+                    listClassName="u-no-margin--bottom"
+                    noBorder
+                  />
+                </Col>
+              </Row>
+            </>
+          }
+        >
+          <Route exact path="/machines" component={MachineList} />
+          <Route exact path="/pools" component={Pools} />
+        </Section>
+      </Route>
+      <Route exact path="/machines/add-machine" component={AddMachine} />
+      <Route exact path="/machines/add-chassis" component={AddChassis} />
```

Comment:
How about these routes instead?

`/machines/add` and `/machines/chassis/add`

Reply:
Might be odd if /machines/chassis is not a destination.

Reply:
Might be odd if /machines/chassis is not a destination.

Reply:
Maybe `/machines/add` and `/machines/add-chassis`? A chassis isn't really a *thing*, but rather a convenience for adding multiple VMs at once so I think it probably shouldn't get its own subpath.